util/app/discovery/discovery.go

```diff
@@ -174,7 +174,7 @@ func cmpSupports(ctx context.Context, pluginSockFilePath, appPath, repoPath, fil
 	}
 
 	// if discovery is not configured, return the client without further checks
-	if !cfg.IsDiscoveryConfigured {
+	if !cfg.IsDiscoveryConfigured && namedPlugin {
 		return conn, cmpClient, true
 	}
```

Comment:
```suggestion
	if !cfg.IsDiscoveryConfigured {
		// If discovery isn't configured but the plugin is named, then the plugin supports the repo.
		if namedPlugin {
			return conn, cmpClient, true
		}
		return nil, nil, false
	}
```

Your change would cause us to call `matchRepositoryCMP` in the case where discovery isn't configured and the plugin isn't named. But it's nonsensical to call `matchRepositoryCMP` if discovery isn't configured. 

This changes makes sure we never unnecessarily call `matchRepositoryCMP`. And it also ensures that we return `true` in the case that discovery isn't configured but the plugin is named.

Reply:
Make sense! I pushed the suggestion.